[docs] R in Gitpod (#480)
gitpod-io/website: src/docs/r-in-gitpod.md

# R in Gitpod

> In order to work with R in Gitpod, you will need to properly configure your project. Here is how to do it.

Reviewer: I know I suggested "project", but this may be ambiguous. "Repository" may be better:

```suggestion
In order to work with R in Gitpod, you will need to properly configure your repository. Here is how to do it.
```

## Installing R

First you'll probably want to install R itself. To do this, add a new file to your repository called [`.gitpod.Dockerfile`](https://www.gitpod.io/docs/config-docker/), and add the following content to it:

```Dockerfile
FROM gitpod/workspace-full

RUN brew install R
```

> Next, create a Gitpod configuration file called `.gitpod.yml` (you can also simply run `gp init` and Gitpod will generate the file), and reference the above Dockerfile in it like so:

Reviewer: Good point, I forgot about `gp init`. Maybe the entire flow of "add this to your dockerfile" and "add this to your .gitpod.yml" could be better rewritten based on `gp init`?

I.e. here, users manually create a `.gitpod.Dockerfile`, but then they can try `gp init`? Would be much simpler to do it the other way around.

Reviewer: Alternatively, instead of `gp init`, users could also try the automated Setup tool in `F1` > `Setup Assistant`? You can choose which one seems best. 🙂

Author: I think gp init is the best way to go, because we have no setup for R

```yml
image:
  file: .gitpod.Dockerfile
```

Then commit and push both files to your repository. After that, every new Gitpod workspace created for your project will have R pre-installed.

## VSCode Extensions

### R Extension

![Picture of R extension](images/RGitpod.png)
To install the R extension to your repository, add the following to your [`.gitpod.yml`](https://www.gitpod.io/docs/config-gitpod-file/) file:

```yaml
vscode:
  extensions:
    - Ikuyadeu.r@1.2.1:z5vr1v1bfS++U/aHLSXQ6Q==
```

> Here is a minimal example of R running in Gitpod, try it!

Reviewer: Maybe a bit too forceful? 😅 

```suggestion
Here is a minimal example of [R running in Gitpod](https://github.com/gitpod-io/Gitpod-R). You can try it here:
```

Author: Sorry, 😅

# R in Gitpod

> Gitpod does not come with out of the box support for R, so here is how to configure it

Reviewer: ```suggestion
In order to work with R in Gitpod, you will need to properly configure your project. Here is how to do it.
```

## Installing R

> First you'll probably want to install R, to do this add the following to your [`.gitpod.Dockerfile`](https://www.gitpod.io/docs/config-docker/)

Reviewer: Most people reading these documents will not already have a Dockerfile. Please assume the reader is 100% new to Gitpod and just trying it out.

```suggestion
First you'll probably want to install R itself. To do this, add a new file to your repository called [`.gitpod.Dockerfile`](https://www.gitpod.io/docs/config-docker/), and add the following content to it:
```

Author: We might also add something like 'or run `gp init`'

```Dockerfile

> RUN brew install R

Reviewer: ```suggestion
FROM gitpod/workspace-full

RUN brew install R
```

Author: Were are asking them to add this to their file possibly making them also add the `FROM` statement aswell

Author: nvm we should assume this is a new setup

```

> This will install R to your project!

Reviewer: ```suggestion
Next, create a Gitpod configuration file called `.gitpod.yml`, and reference the above Dockerfile in it like so:

```yml
image:
  file: .gitpod.Dockerfile
```

Then commit and push both files to your repository. After that, every new Gitpod workspace created for your project will have R pre-installed.
```

## VSCode Extensions

### R Extension
![Picture of R extension](images/RGitpod.png)

> To install the R extension to your repository add the following to your [`.gitpod.yml`](https://www.gitpod.io/docs/config-gitpod-file/):

Reviewer: ```suggestion
To install the R extension to your repository, add the following to your [`.gitpod.yml`](https://www.gitpod.io/docs/config-gitpod-file/) file:
```
